pkg/controller/deployment/deployment_controller.go

```diff
@@ -114,23 +353,22 @@ func (d *DeploymentController) getTargetedRCs(deployment *experimental.Deploymen
 
 // Returns an RC that matches the intent of the given deployment.
 // It creates a new RC if required.
-func (d *DeploymentController) getDesiredRC(deployment *experimental.Deployment) (*api.ReplicationController, error) {
+func (dc *DeploymentController) getDesiredRC(deployment *experimental.Deployment) (*api.ReplicationController, error) {
 	namespace := deployment.ObjectMeta.Namespace
 	// Find if the required RC exists already.
-	rcList, err := d.client.ReplicationControllers(namespace).List(labels.Everything())
+	rcList, err := dc.rcStore.List()
 	if err != nil {
 		return nil, fmt.Errorf("error listing replication controllers: %v", err)
 	}
-	for _, rc := range rcList.Items {
+	for _, rc := range rcList {
 		if api.Semantic.DeepEqual(rc.Spec.Template, deployment.Spec.Template) {
```

Comment:
This will fail now that we are adding an addition label (uniqueLabelKey) to the created RC.